src/bfs.c

```diff
@@ -0,0 +1,49 @@
+#include "bfs.h"
+
+// Function to initialise the attributes of each node
+struct attributes* init_attributes(struct attributes *nodes, int n) {
+	int i;
+	for (i = 1; i <= n; i++) {
+		nodes[i].parent = NIL;
+		nodes[i].distance = INF;
+		nodes[i].current_color = WHITE;
+	}
+}
+
+// Function to print the value of attributes
+void print_attributes(struct attributes *nodes, int n) {
```

Comment:
I don't think `nodes` is a good name for an array of struct of node attributes. What about `attributes_list` or `node_attributes` ?